src/main/java/org/folio/rest/service/AccountValidationService.java

```diff
@@ -0,0 +1,77 @@
+package org.folio.rest.service;
+
+import javax.ws.rs.core.Response;
+
+import org.folio.rest.jaxrs.model.AccountsCheckRequest;
+import org.folio.rest.jaxrs.model.AccountsCheckResponse;
+import org.folio.rest.jaxrs.resource.Accounts;
+import org.folio.rest.repository.AccountRepository;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+
+public class AccountValidationService {
+
+  private final AccountRepository accountRepository;
+
+  public AccountValidationService(AccountRepository accountRepository) {
+    this.accountRepository = accountRepository;
+  }
+
+  public void validatePayment(String accountId, AccountsCheckRequest entity,
+    Handler<AsyncResult<Response>> asyncResultHandler) {
+
+    accountRepository.getAccountById(accountId)
+      .onComplete(account -> {
```

Comment:
Add error handling and throw 500 in case of failure

Reply:
done